### ISSUE TITLE
MangoHud crashes on recent AMD CPUs with 5.4 kernel

### ISSUE THREAD

**Issue by Emanem** [flightlessmango/MangoHud#440]
Hi,

Looks like MangoHud is trying to access the `k10temp` file from method/function [init_cpu_power_data_k10temp](https://github.com/flightlessmango/MangoHud/blob/0ce22ddc15ff61e27b9c2a1a2b34f6816b430f7f/src/cpu.cpp#L495), unfortunately this yields to a crash on kernels 5.4.xxx (Ubuntu LTS) which I believe don't fully export the _zen3_ _k10temp_ info.

The stack is:

```
Using host libthread_db library "/lib/x86_64-linux-gnu/libthread_db.so.1".
skipping config: /usr/bin/MangoHud.conf [ not found ]
skipping config: /home/ema/.config/MangoHud/vkcube.conf [ not found ]
skipping config: /home/ema/.config/MangoHud/MangoHud.conf [ not found ]
    
Program received signal SIGSEGV, Segmentation fault.
0x00007ffff7c60f5b in _IO_new_fclose (fp=0x326e6f6d77682f) at iofclose.c:48
48	iofclose.c: No such file or directory.
(gdb) bt 10
#0  0x00007ffff7c60f5b in _IO_new_fclose (fp=0x326e6f6d77682f) at iofclose.c:48
#1  0x00007fffef103c3e in init_cpu_power_data_k10temp(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >) ()
   from /home/ema/MangoHud/lib/mangohud/lib64/libMangoHud.so
#2  0x00007fffef10470c in CPUStats::InitCpuPowerData() () from /home/ema/MangoHud/lib/mangohud/lib64/libMangoHud.so
#3  0x00007fffef0fb37f in overlay_CreateInstance(VkInstanceCreateInfo const*, VkAllocationCallbacks const*, VkInstance_T**) () from /home/ema/MangoHud/lib/mangohud/lib64/libMangoHud.so
#4  0x00007ffff7df75d1 in ?? () from /usr/lib/x86_64-linux-gnu/libvulkan.so.1
#5  0x00007ffff7dfb5d1 in vkCreateInstance () from /usr/lib/x86_64-linux-gnu/libvulkan.so.1
#6  0x000055555555843b in ?? ()
#7  0x00007ffff7c030b3 in __libc_start_main (main=0x555555557e80, argc=1, argv=0x7fffffffde88, init=<optimised out>, fini=<optimised out>, rtld_fini=<optimised out>, 
    stack_end=0x7fffffffde78) at ../csu/libc-start.c:308
#8  0x000055555555a81e in ?? ()
```

generated with `MANGOHUD=1 gdb vkcube`.
Kernel `Linux scv 5.4.0-58-generic #64-Ubuntu SMP Wed Dec 9 08:16:25 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux`

**Comment by flgx16** [comment]
What version of MangoHud are you using? It should be fixed in 0.6.1.

**Comment by Emanem** [comment]
Can confirm is working now, can be closed.
